From 3fe9f4428ae8c4a8772218b7abf30b1e642b5b2c Mon Sep 17 00:00:00 2001
From: Jan Kiszka <jan.kiszka@siemens.com>
Date: Fri, 30 Jul 2021 08:59:18 +0200
Subject: [PATCH 26/26] arm64: dts: ti: k3-am65-mcu: Switch to upstream
 watchdog

Bind to the backported rti_wdt, rather than the legacy SDK version. Also
enable shared power control.  This is needed when k3-rti-wdt is used as
firmware and locked the power control for this IP block.

Signed-off-by: Jan Kiszka <jan.kiszka@siemens.com>
---
 arch/arm64/boot/dts/ti/k3-am65-mcu.dtsi | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/ti/k3-am65-mcu.dtsi b/arch/arm64/boot/dts/ti/k3-am65-mcu.dtsi
index 450456460182..605be5abea81 100644
--- a/arch/arm64/boot/dts/ti/k3-am65-mcu.dtsi
+++ b/arch/arm64/boot/dts/ti/k3-am65-mcu.dtsi
@@ -441,10 +441,10 @@
 	};
 
 	mcu_rti1: rti@40610000 {
-		compatible = "ti,keystone-wdt";
+		compatible = "ti,j7-rti-wdt";
 		reg = <0x0 0x40610000 0x0 0x100>;
 		clocks = <&k3_clks 135 0>;
-		power-domains = <&k3_pds 135 TI_SCI_PD_EXCLUSIVE>;
+		power-domains = <&k3_pds 135 TI_SCI_PD_SHARED>;
 		assigned-clocks = <&k3_clks 135 0>;
 		assigned-clock-parents = <&k3_clks 135 4>;
 	};
-- 
2.31.1

